compute/compute-node.Dockerfile

```diff
@@ -77,9 +77,6 @@
 # build_and_test.yml github workflow for how that's done.
 
 ARG PG_VERSION
-ARG REPOSITORY=ghcr.io/neondatabase
-ARG IMAGE=build-tools
-ARG TAG=pinned
```

Comment:
Also need to remove it from the workflow: 
- https://github.com/neondatabase/neon/blob/f4c1bed6c908b63aae31c5f5d857d80ded342548/.github/workflows/build_and_test.yml#L746

- https://github.com/neondatabase/neon/blob/f4c1bed6c908b63aae31c5f5d857d80ded342548/.github/workflows/build_and_test.yml#L766

And we can remove `build-build-tools-image` dependency from here:
- https://github.com/neondatabase/neon/blob/f4c1bed6c908b63aae31c5f5d857d80ded342548/.github/workflows/build_and_test.yml#L672-L673